docs/guides/using-the-network-shim.md

```diff
@@ -0,0 +1,193 @@
+# Using the Network Shim
+
+## About
+
+The Network Shim is a tool that provides a cypress plugin as well as a setup utilility for cypress' `setupFile`, which have to be used in conjunction. It uses the cypress `cy.intercept()` API to intercept network traffic during a test run. It has three different _modes_:
+
+-   `capture`: when operating in this mode, the Network Shim will intercept all outgoing requests to the specified hosts and allow each request to reach the destination server. Each response is stored as a fixture (JSON file).
+-   `stub`: in this mode the Network Shim also intercepts all outgoing requests to the specified hosts, but it will prevent the request from reaching the destination server and instead it will produce a response from a previously recorded fixture.
+-   `live`: the Network Shim does not intercept any requests when this mode is enabled.
+
+By first recording fixtures during a _capture run_ and then using these fixtures during a _stub run_ it effectively becomes possible to execute a full cypress test suite without a live backend. This is especially relevant for running a cypress test suite on CI.
+
+A capture or stub run is executed for a specific DHIS2 api version and individual fixtures for each version are produced/used during a run. This paves the way for doing cross-version testing for _feature-toggling_ apps.
+
+## General workflow
+
+As mentioned, he Network Shim is a tool that enables a workflow where a cypress e2e test suite is executed on CI.
+
+The general steps required to setup a workflow like that are as follows:
```

Comment:
Maybe for the workflow part of these steps we can refer to our workflows repo.

Reply:
Also I guess people don't necessarily have to use GH actions. We could mention that that's what we use, but you can use any CI you'd like.